openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -591,8 +591,11 @@ def login_user(request, api_version='v1'):
 
         if third_party_auth_requested:
             running_pipeline = pipeline.get(request)
-            finish_auth_url = pipeline.get_complete_url(backend_name=running_pipeline['backend'])
-
+            if running_pipeline:
+                finish_auth_url = pipeline.get_complete_url(backend_name=running_pipeline['backend'])
+            else:
+                log.warning("Unable to find finish_auth_url for user %s. Using default of ''.", user.id) 
```

Comment:
**Author Note:** Should we also enhance the except with a catch-all that logs an error about unknown login exceptions and then rethrows it? This potentially could have lead to more easily noticing this uncaught TypeError. Additionally, a custom attribute could be added so Vanguard's could alert on this case. This could be done on an earlier PR before this issue is fixed.